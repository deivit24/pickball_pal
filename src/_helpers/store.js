import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
