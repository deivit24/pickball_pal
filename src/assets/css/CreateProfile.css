#form-box-create {
  width: 100%;
  padding: 20px;
  margin-top: 10%;
  box-shadow: 0 0 25px 3px #afafaf;
  border-radius: 25px;
  z-index: 2;
}

#form-box-create .form-title {
  background: #3f50b5;
  color: white;
  text-align: center;
  padding: 30px 0px;
  position: relative;
  margin-top: -70px;
  margin-bottom: 30px;
  border-radius: 10px;
  z-index: 20;
}

#form-box-create .social-button {
  margin-top: 20px;
  margin-right: 10px;
}
#form-box-create .submit-button {
  float: left !important;
}
#form-box-create p {
  margin-top: 20px;
}
