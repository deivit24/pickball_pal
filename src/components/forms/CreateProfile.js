// React
import React, { useState } from 'react';
// Redux
import { useDispatch } from 'react-redux';
// React Router DOM
import { useHistory, Link } from 'react-router-dom';
// Create Profile Action
import { createProfile } from '../../_actions/profile';

import {
  TextField,
  MenuItem,
  OutlinedInput,
  InputLabel,
  InputAdornment,
  FormControl,
  Button,
} from '@material-ui/core';

import '../../assets/css/CreateProfile.css';

const CreateProfile = () => {
  const [displaySocials, toggleDisplaySocials] = useState(false);
  const history = useHistory();
  const dispatch = useDispatch();

  const INIT_STATE = {
    gender: '',
    level: '',
    bio: '',
    active: false,
    playing_tyle: '',
    zip: '',
    facebook: '',
    instagram: '',
    linkedin: '',
  };
  const [values, setValues] = useState(INIT_STATE);

  const {
    gender,
    level,
    active,
    bio,
    zip,
    playing_tyle,
    facebook,
    instagram,
    linkedin,
  } = values;
  const [activeToggle, setActiveToggle] = useState(false);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setValues((l) => ({ ...l, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(createProfile(values, history));
  };
  return (
    <div id="CreateProfile" className="container">
      <div className="form-box-create">
        <div className="form-title">
          <h2>Profile</h2>
        </div>
        <form noValidate autoComplete="on">
          <div className="row">
            <div className="col-md-6">
              <TextField
                fullWidth
                margin="normal"
                variant="outlined"
                label="Gender"
                id="select"
                value={gender}
                select
                name="gender"
                onChange={handleChange}
              >
                <MenuItem value="">
                  <em>None</em>
                </MenuItem>
                <MenuItem value="male">Male</MenuItem>
                <MenuItem value="female">Female</MenuItem>
                <MenuItem value="non-binary">Non-Binary</MenuItem>
              </TextField>
            </div>
            <div className="col-md-6">
              <TextField
                fullWidth
                variant="outlined"
                label="Level"
                id="select"
                value={level}
                select
                margin="normal"
                name="level"
                onChange={handleChange}
              >
                <MenuItem value="I don't know">
                  <em>I don't know</em>
                </MenuItem>
                <MenuItem value="1">1.0</MenuItem>
                <MenuItem value="1.5">1.5</MenuItem>
                <MenuItem value="2.0">2.0</MenuItem>
                <MenuItem value="2.5">2.5</MenuItem>
                <MenuItem value="3.0">3.0</MenuItem>
                <MenuItem value="3.5">3.5</MenuItem>
                <MenuItem value="4.0">4.0</MenuItem>
                <MenuItem value="4.5">4.5</MenuItem>
                <MenuItem value="5.0">5.0</MenuItem>
              </TextField>
            </div>
            <div className="col-md-6">
              <TextField
                fullWidth
                variant="outlined"
                label="Active"
                id="select"
                value={active}
                select
                helperText="Want your profile to be visible?"
                margin="normal"
                name="active"
                onChange={(e) => {
                  setValues({ ...values, active: !active });
                  setActiveToggle(!activeToggle);
                }}
              >
                <MenuItem value={true}>Yes</MenuItem>
                <MenuItem value={false}>No</MenuItem>
              </TextField>
            </div>
            <div className="col-md-6">
              <TextField
                id="outlined-textarea"
                label="Zip Code"
                name="zip"
                type="number"
                value={zip}
                onChange={handleChange}
                helperText="e.g. 78745"
                fullWidth
                margin="normal"
                variant="outlined"
              />
            </div>
            <div className="col-md-6">
              <TextField
                id="outlined-textarea"
                label="Tell us a little about yourself"
                name="bio"
                type="text"
                value={bio}
                onChange={handleChange}
                helperText=""
                multiline
                rows={4}
                fullWidth
                margin="normal"
                variant="outlined"
              />
            </div>

            <div className="col-md-6">
              <TextField
                id="outlined-textarea"
                label="What is your playing style like?"
                name="playing_tyle"
                type="text"
                value={playing_tyle}
                onChange={handleChange}
                helperText=""
                multiline
                rows={4}
                fullWidth
                margin="normal"
                variant="outlined"
              />
            </div>
            <div className="col-12">
              <Button
                onClick={() => toggleDisplaySocials(!displaySocials)}
                variant="outlined"
                size="small"
                color="primary"
                className="social-button float-left"
              >
                Enter Social Accounts
              </Button>
              <p> (Optional)</p>
            </div>
            {displaySocials && (
              <>
                <div className="col-md-4">
                  <FormControl variant="outlined" margin="normal">
                    <InputLabel htmlFor="facebook">Facebook URL</InputLabel>
                    <OutlinedInput
                      id="facebook"
                      type="text"
                      value={facebook.split(' ').join('')}
                      name="facebook"
                      onChange={handleChange}
                      startAdornment={
                        <InputAdornment position="start">
                          <i className="fab fa-facebook"></i>
                        </InputAdornment>
                      }
                      labelWidth={120}
                    />
                  </FormControl>
                </div>
                <div className="col-md-4">
                  <FormControl variant="outlined" margin="normal">
                    <InputLabel htmlFor="instagram">Instagram URL</InputLabel>
                    <OutlinedInput
                      id="instagram"
                      type="text"
                      value={instagram.split(' ').join('')}
                      name="instagram"
                      onChange={handleChange}
                      startAdornment={
                        <InputAdornment position="start">
                          <i className="fab fa-instagram"></i>
                        </InputAdornment>
                      }
                      labelWidth={120}
                    />
                  </FormControl>
                </div>
                <div className="col-md-4">
                  <FormControl variant="outlined" margin="normal">
                    <InputLabel htmlFor="linkedin">Linkedin URL</InputLabel>
                    <OutlinedInput
                      id="linkedin"
                      type="text"
                      value={linkedin.split(' ').join('')}
                      name="linkedin"
                      onChange={handleChange}
                      startAdornment={
                        <InputAdornment position="start">
                          <i className="fab fa-linkedin"></i>
                        </InputAdornment>
                      }
                      labelWidth={120}
                    />
                  </FormControl>
                </div>
              </>
            )}
            <Link className="submit-button" to="/dashboard">
              <Button
                // onClick={handleSubmit}
                variant="contained"
                color="secondary"
              >
                Go Back
              </Button>
            </Link>

            <Button
              onClick={handleSubmit}
              variant="contained"
              color="primary"
              className="submit-button"
            >
              Submit
            </Button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default CreateProfile;
