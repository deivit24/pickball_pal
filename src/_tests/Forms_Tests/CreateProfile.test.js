import React from 'react';
import { render } from '@testing-library/react';
import CreateProfile from '../../components/forms/CreateProfile';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing CreateProfile

describe('Testing the CreateProfile', () => {
  it('it renders CreateProfile without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <CreateProfile />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CreateProfile />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
