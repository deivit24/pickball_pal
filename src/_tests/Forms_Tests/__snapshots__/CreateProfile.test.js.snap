// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CreateProfile matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    id="CreateProfile"
  >
    <div
      class="form-box-create"
    >
      <div
        class="form-title"
      >
        <h2>
          Profile
        </h2>
      </div>
      <form
        autocomplete="on"
        novalidate=""
      >
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="select"
                id="select-label"
              >
                Gender
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="select-label select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="select"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="gender"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Gender
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="select"
                id="select-label"
              >
                Level
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="select-label select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="select"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="level"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Level
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink="true"
                for="select"
                id="select-label"
              >
                Active
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="select-label select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="select"
                  role="button"
                  tabindex="0"
                >
                  No
                </div>
                <input
                  aria-describedby="select-helper-text"
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="active"
                  tabindex="-1"
                  value="false"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
                  >
                    <span>
                      Active
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                id="select-helper-text"
              >
                Want your profile to be visible?
              </p>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-textarea"
                id="outlined-textarea-label"
              >
                Zip Code
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-describedby="outlined-textarea-helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-textarea"
                  name="zip"
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Zip Code
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained"
                id="outlined-textarea-helper-text"
              >
                e.g. 78745
              </p>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-textarea"
                id="outlined-textarea-label"
              >
                Tell us a little about yourself
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                  id="outlined-textarea"
                  name="bio"
                  rows="4"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Tell us a little about yourself
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-textarea"
                id="outlined-textarea-label"
              >
                What is your playing style like?
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                  id="outlined-textarea"
                  name="playing_tyle"
                  rows="4"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      What is your playing style like?
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="col-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined social-button float-left MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Enter Social Accounts
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <p>
               (Optional)
            </p>
          </div>
          <a
            class="submit-button"
            href="/dashboard"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Go Back
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained submit-button MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
